lib/Cake/View/Helper.php

```diff
@@ -552,7 +552,7 @@ protected function _formatAttribute($key, $value, $escape = true) {
  */
 	protected function _confirm($message, $okCode, $cancelCode = '', $options = array()) {
 		$message = json_encode($message);
-		$confirm = "if (confirm({$message})) { {$okCode} } {$cancelCode}";
+		$confirm = "if (confirm({$message})) { {$okCode} } else { {$cancelCode} }";
```

Comment:
Won't this change the behavior of `_confirm` for user-land helpers too? The call 

```php
$this->_confirm('are you sure?', 'doYes()', 'doNo()')
```

Previously would make the following:

```js
if (confirm('are you sure?')) {
  doYes()
}
doNo();
```

But now generates:

```js
if (confirm('are you sure?')) {
  doYes()
} else {
  doNo();
}
```

That isn't going to do the same thing, which makes me think this change isn't backwards compatible.